clang build failed:
2026-03-10T10:25:27.7766257Z drivers/spi/spi-phytium-v2.c:143:6: error: variable 'nor' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
2026-03-10T10:25:27.7805028Z   143 |         if (mem)
2026-03-10T10:25:27.7844503Z       |             ^~~
2026-03-10T10:25:27.7864784Z drivers/spi/spi-phytium-v2.c:161:13: note: uninitialized use occurs here
2026-03-10T10:25:27.7866230Z   161 |         if (mem == nor->spimem && fts->tx && fts->len == 1) {
2026-03-10T10:25:27.7867180Z       |                    ^~~
2026-03-10T10:25:27.7868366Z drivers/spi/spi-phytium-v2.c:143:2: note: remove the 'if' if its condition is always true
2026-03-10T10:25:27.7869425Z   143 |         if (mem)
2026-03-10T10:25:27.7870026Z       |         ^~~~~~~~
2026-03-10T10:25:27.7870744Z   144 |                 nor = spi_mem_get_drvdata(mem);
2026-03-10T10:25:27.7871987Z drivers/spi/spi-phytium-v2.c:140:21: note: initialize the variable 'nor' to silence this warning
2026-03-10T10:25:27.7873489Z   140 |         struct spi_nor *nor;
2026-03-10T10:25:27.7874167Z       |                            ^
2026-03-10T10:25:27.7874927Z       |                             = NULL
2026-03-10T10:25:27.7954864Z 1 error generated.
2026-03-10T10:25:27.8135146Z make[4]: *** [scripts/Makefile.build:243：drivers/spi/spi-phytium-v2.o] 错误 1
2026-03-10T10:25:27.8189229Z make[3]: *** [scripts/Makefile.build:480：drivers/spi] 错误 2
2026-03-10T10:25:27.8190291Z make[3]: *** 正在等待未完成的任务....

--- a/drivers/spi/spi-phytium-plat-v2.c
+++ b/drivers/spi/spi-phytium-plat-v2.c
@@ -26,7 +26,7 @@
 #include "spi-phytium.h"
 
 #define DRIVER_NAME_PHYT "phytium_spi_2.0"
-#define DRIVER_VERSION	"1.0.7"
+#define DRIVER_VERSION	"1.0.8"
 
 #define PHYTIUM_CPU_PART_FTC872		0x872
 
@@ -130,7 +130,7 @@ static int spi_phyt_probe(struct platform_device *pdev)
 	int ret;
 	int num_cs;
 	int cs_gpio;
-	int global_cs = 0;
+	int global_cs = 1;
 	int i;
 	u32 clk_rate = SPI_DEFAULT_CLK;
 

--- a/drivers/spi/spi-phytium-v2.c
+++ b/drivers/spi/spi-phytium-v2.c
@@ -15,6 +15,8 @@
 #include <linux/platform_device.h>
 #include <linux/slab.h>
 #include <linux/spi/spi.h>
+#include <linux/spi/spi-mem.h>
+#include <linux/mtd/spi-nor.h>
 #include <linux/scatterlist.h>
 #include <linux/module.h>
 #include <linux/of.h>
@@ -24,10 +26,6 @@
 #include <linux/acpi.h>
 #include <linux/mtd/spi-nor.h>
 #include "spi-phytium.h"
-
-#define MCP251x_READ		0x03
-#define MCP251x_READ_RXB0	0x90
-#define MCP251x_READ_RXB1	0x94
 
 static inline void spi_phyt_enable_chip(struct phytium_spi *fts, u8 enable)
 {
@@ -99,13 +97,6 @@ static void spi_phyt_set_cs(struct spi_device *spi, bool enable)
 	u32 origin;
 	u16 cs;
 
-	if (fts->tx || fts->rx)
-		return;
-
-	if (fts->msg->cmd_id == PHYTSPI_MSG_CMD_DATA &&
-			fts->msg->cmd_subid == PHYTSPI_MSG_CMD_DATA_TX)
-		return;
-
 	if (chip && chip->cs_control)
 		chip->cs_control(!enable);
 
@@ -145,7 +136,12 @@ static int spi_phyt_transfer_one(struct spi_master *master,
 {
 	struct phytium_spi *fts = spi_master_get_devdata(master);
 	struct chip_data *chip = spi_get_ctldata(spi);
+	struct spi_mem *mem = spi_get_drvdata(spi);
+	struct spi_nor *nor = NULL;
 	int ret;
+
+	if (mem)
+		nor = spi_mem_get_drvdata(mem);
 
 	fts->tx = (void *)transfer->tx_buf;
 	fts->tx_end = fts->tx + transfer->len;
@@ -162,7 +158,7 @@ static int spi_phyt_transfer_one(struct spi_master *master,
 			chip->tmode = TMOD_TO;
 	}
 
-	if (fts->tx && fts->len == 1) {
+	if (mem == nor->spimem && fts->tx && fts->len == 1) {
 		if ((*(u8 *)fts->tx == SPINOR_OP_WREN) && fts->spi_write_flag == 0) {
 			spi_phytium_write_pre(fts, spi->chip_select,
 					transfer->bits_per_word, spi->mode,
@@ -262,11 +258,7 @@ static int spi_phyt_transfer_one(struct spi_master *master,
 			}
 			fts->flash_erase = 0;
 		} else {
-			fts->flags = 1;
-			if (fts->spi_write_flag == 0 && *(u8 *)(fts->tx) != MCP251x_READ
-					&& *(u8 *)(fts->tx) != MCP251x_READ_RXB0
-					&& *(u8 *)(fts->tx) != MCP251x_READ_RXB1)
-				fts->flags = 3;
+			fts->flags = 0;
 
 			ret = spi_phytium_write(fts, spi->chip_select, transfer->bits_per_word,
 					spi->mode, chip->tmode, fts->flags, fts->spi_write_flag);

